server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -984,23 +1031,23 @@ private long balanceTablets() {
             continue;
           }
           migrationsOutForLevel++;
-          migrations.put(ke, TabletServerIdImpl.toThrift(m.getNewTabletServer()));
+          getContext().getAmple().mutateTablet(ke)
+              .putMigration(TabletServerIdImpl.toThrift(m.getNewTabletServer())).mutate();
           log.debug("migration {}", m);
         }
         totalMigrationsOut += migrationsOutForLevel;
 
         // increment this at end of loop to signal complete run w/o any continue
         levelsCompleted++;
       }
-      balancerMetrics.assignMigratingCount(migrations::size);
+      balancerMetrics.assignMigratingCount(Manager.this::numMigrations);
```

Comment:
This will scan the metadata table again. It was scanned earlier, wonder if we can use the map and a count of new migrations we added here to get the new count.  Not sure if that will work, may be something to look into.

Reply:
Calculate total from existing values instead of rescanning the metadata